src/apps/api/serializers/submissions.py

```diff
@@ -105,24 +109,32 @@ class Meta:
     def get_filename(self, instance):
         return basename(instance.data.data_file.name)
 
-    # TODO: Validate the user is a participant in this competition.phase
-
     def create(self, validated_data):
-        validated_data["owner"] = self.context['owner']
         sub = super().create(validated_data)
         sub.start()
         return sub
 
     def validate(self, attrs):
         data = super().validate(attrs)
+
+        if not self.instance:
+            is_in_competition = data["phase"].competition.participants.filter(
+                user=self.context["request"].user,
+                status=CompetitionParticipant.APPROVED
+            ).exists()
+            if not is_in_competition:
+                raise PermissionDenied("You do not have access to this competition to make a submission")
+
+        # TODO: Explain what this is doing? I think it's setting task_pk for nested writable stuff to work well?
```

Comment:
we are setting task_pk to get picked up again on line 161. so we know which task the submission was running against and we can fire off the scoring step with the same task.

Reply:
ty